src/Monthpicker.js

```diff
@@ -129,59 +116,46 @@ class MonthPicker extends React.Component {
   }
 
   handleKeyDown = event => {
-    const { focussedDate, year } = this.state
+    const { open } = this.state
+    const { year, month } = this.props
 
-    const focussedYear = getYear(focussedDate)
-    const matchesCurrentYear = focussedYear === year
+    if (!open) return
+
+    const currentDate = new Date(year, month - 1)
 
     const firstMonthOfSelectedYear = new Date(year, 0)
 
     switch (event.key) {
       case 'ArrowLeft': {
         event.preventDefault()
 
-        const nextFocussedDate = matchesCurrentYear ? subMonths(focussedDate, 1) : firstMonthOfSelectedYear
-
-        if (getYear(nextFocussedDate) < year) this.previousYear()
-
-        return this.setFocussedMonth(nextFocussedDate)
+        const nextDate = subMonths(currentDate, 1)
+        return this.changeValue(nextDate, event)
       }
 
       case 'ArrowRight': {
         event.preventDefault()
 
-        const nextFocussedDate = matchesCurrentYear ? addMonths(focussedDate, 1) : firstMonthOfSelectedYear
-
-        if (getYear(nextFocussedDate) > year) this.nextYear()
-
-        return this.setFocussedMonth(nextFocussedDate)
+        const nextDate = addMonths(currentDate, 1)
+        return this.changeValue(nextDate, event)
       }
 
       case 'ArrowUp': {
         event.preventDefault()
 
-        const nextFocussedDate = matchesCurrentYear ? subMonths(focussedDate, 3) : firstMonthOfSelectedYear
-
-        if (getYear(nextFocussedDate) < year) this.previousYear()
-
-        return this.setFocussedMonth(nextFocussedDate)
+        const nextDate = subMonths(currentDate, 3)
+        return this.changeValue(nextDate, event)
       }
 
       case 'ArrowDown': {
         event.preventDefault()
```

Comment:
Never reviewed this before, but can't this be just put at the top over the `switch` statement once?

Reply:
No!

Reply:
:okeh:

Reply:
(I only want to prevent the default for arrow keys and enter)